src/pages/studyView/StudyViewPage.tsx

```diff
@@ -4,71 +4,138 @@ import * as _ from 'lodash';
 import $ from 'jquery';
 import {observer} from "mobx-react";
 import {remoteData} from "../../shared/api/remoteData";
-import {action, computed, observable, reaction} from "mobx";
-import client from "shared/api/cbioportalInternalClientInstance";
-
-
-function getMutatedGenes(query:string){
-    console.log("fetching genes");
-    return new Promise((resolve,reject)=>{
-        setTimeout(function(){
-            resolve(_.filter(["aaron","nora","natasha", "john"],(name:string)=>name.includes(query)))
-        },1000);
-    });
-
-}
-
-
-// this is an example of using client.
-client.getAllGenesetsUsingGET({}).then(function(data){
-    console.log(data);
-});
-
-class StudyViewPageStore {
+import {action, computed, observable, reaction, ObservableMap} from "mobx";
+import internalClient from "shared/api/cbioportalInternalClientInstance";
+import defaultClient from "shared/api/cbioportalClientInstance";
+import { ClinicalDataCountFilter, StudyViewFilter, ClinicalDataCount, ClinicalDataEqualityFilter } from 'shared/api/generated/CBioPortalAPIInternal';
+import { PieChart } from './charts/pieChart/PieChart'
+import { ClinicalAttribute } from 'shared/api/generated/CBioPortalAPI';
+import styles from "./styles.module.scss";
+
+export type ClinicalDataType= "SAMPLE" | "PATIENT";
+export type ClinicalAttributeData = {[attrId:string]:ClinicalDataCount[]};
+export type ClinicalAttributeDataWithMeta = {attributeId:string,clinicalDataType:ClinicalDataType,counts:ClinicalDataCount[]};
+export class StudyViewPageStore {
 
     constructor(){
 
     }
 
-    @observable filterTerm = "";
-
-    readonly mutatedGenes = remoteData({
-
-        await:()=>[
-            this.prelimData
-        ],
-
-        invoke:()=>{
-
-            return getMutatedGenes(this.filterTerm);
-
+    @observable studyId = "hnsc_tcga";
+
+    @observable sampleAttrIds = ["OCT_EMBEDDED"];
+
+    @observable patientAttrIds = ["AJCC_PATHOLOGIC_TUMOR_STAGE","ICD_O_3_HISTOLOGY"];
+
+    @observable private _clinicalDataEqualityFilter = observable.map<ClinicalDataEqualityFilter>();
+
+    @action updateClinicalDataEqualityFilters( attributeId      : string,
+                                               clinicalDataType : ClinicalDataType,
+                                               value            : string) {
+        
+        let id = [clinicalDataType,attributeId].join('_');
+
+        let clinicalDataEqualityFilter =this._clinicalDataEqualityFilter.get(id);
```

Comment:
this should be clinicalDataEqalityFilterMap or collection or something.  it needs to reflect the fact that there are multiple clinicalDataEqualityFilter inside of it.